game-core/src/main/java/games/strategy/ui/ImageScrollerLargeView.java

```diff
@@ -305,18 +308,16 @@ protected void refreshBoxSize() {
    * @param value The new scale value. Constrained to the bounds of no less than 0.15 and no greater than 1.
    *        If out of bounds the nearest boundary value is used.
    */
-  public void setScale(double value) {
-    if (value < 0.15) {
-      value = 0.15;
-    }
-    if (value > 1) {
-      value = 1;
-    }
+  public void setScale(final double value) {
+    scale = normalizeScale(value);
+    refreshBoxSize();
+  }
+
+  @VisibleForTesting
+  static double normalizeScale(final double value) {
     // we want the ratio to be a multiple of 1/256
     // so that the tiles have integer widths and heights
-    value = ((int) (value * 256)) / ((double) 256);
-    scale = value;
-    refreshBoxSize();
+    return ((int) (Doubles.constrainToRange(value, 0.15, 1.0) * 256.0)) / 256.0;
```

Comment:
I'm kinda confused on why this was not simplified. Seems like all we are doing is contraining a scale value to 0.15 to 1.0, can we just not simply make the call to `Doubles` and return? The multiply by 256 to divide seems like it can be simplified, but it's confusing why we would want to multiple by 256, take the floor of that number, to then divide by 256 which introduces a lot of round-off error.

Reply:
It's existing logic here, so this is strictly not a blocker, but would be good to see this solved so we do not need to revisit this code for that clarity.

Reply:
I think the preceding comment provides the context:

```java
// we want the ratio to be a multiple of 1/256
// so that the tiles have integer widths and heights
```

My guess is it has to do with the current size of a tile (256 x 256 pixels).  I could replace the magic number with `Tile#TILE_SIZE`, but that would create a messy dependency from this package to `g.s.triplea.ui`.  A better solution may be to inject the tile size via the constructor so this view isn't coupled to a 256 x 256 tile size.

Reply:
bc21fb2f3

Reply:
Thanks for the pointer and update. The commentary I think would be ideally folded into the code for self-documenting code, but not sure how to do that in this case. I would agree there is not much more to do to help this code for now. :+1:

Reply:
b2a13a87a

The tests were basically a vise to guard against a regression in the calculation.

Now that the constrain/discretize behaviors are split into two separate methods, the tests probably serve no useful purpose:

* The test for `constrainScale()` would just be testing a Guava method, which is not our responsibility.
* The test for `discretizeScale()`, if not data-driven, would just duplicate the production algorithm to generate the expected values, which is a smell.

So I simply verified the composition of `discretizeScale()` and `constrainScale()` produced the same values as the original code before removing the tests.